duckshoot.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python
+# coding: Latin
+
+import json
+from my_button import MyScale
+# Load all standard tools for image processing challenges
+from img_base_class import *
+
+
+# Image stream processing thread
+class StreamProcessor(threading.Thread):
+    def __init__(self, screen=None, camera=None, drive=None, colour="any"):
+        super(StreamProcessor, self).__init__()
+        self.camera = camera
+        image_width, image_height = self.camera.resolution
+        self.drive = drive
+        self.drive.should_normalise_motor_speed = False
+        self.screen = screen
+        self.stream = picamera.array.PiRGBArray(camera)
+        self.event = threading.Event()
+        self.terminated = False
+        self.MAX_AREA = 2000  # Largest target to shoot at
+        self.MIN_CONTOUR_AREA = 400
+        self.MAX_TARGET_SIZE = 2000
+        self.MAX_TARGET_WIDTH = 80
+        self.AIM_OFFSET = 45
+        self._colour = colour
+        self.found = False
+        self.cycle = 0
+        self.target_number = 0
+        self.menu = False
+        self.last_t_error = 0
+        self.TURN_P = 0.8
+        self.TURN_D = 0.4
+        self.CROP_WIDTH = 320
+        self.CROP_HEIGHT = 60
+        self.image_centre_x = self.CROP_WIDTH / 2.0
+        self.image_centre_y = self.CROP_HEIGHT / 2.0
+        self.CROP_H_OFFSET = 160
+        self.CROP_V_OFFSET = 180
+        self.colour_bounds = json.load(open('duckshoot.json'))
+        self.hsv_lower = (0, 0, 0)
+        self.hsv_upper = (0, 0, 0)
+        self.DRIVING = True
+        self.tracking = False
+        # Why the one second sleep?
+        time.sleep(1)
+        self.start()
+
+    @property
+    def colour(self):
+        """Set the target colour property"""
+        return self._colour
+
+    @colour.setter
+    def colour(self, colour):
+        self._colour = colour
+
+    def run(self):
+        # This method runs in a separate thread
+        while not self.terminated:
+            # Wait for an image to be written to the stream
+            if self.event.wait(1):
+                try:
+                    # Read the image and do some processing on it
+                    self.stream.seek(0)
+                    self.process_image(self.stream.array, self.screen)
+                finally:
+                    # Reset the stream and event
+                    self.stream.seek(0)
+                    self.stream.truncate()
+                    self.event.clear()
+
+    # Image processing function
+    def process_image(self, image, screen):
+        screen = pygame.display.get_surface()
+        # crop image to speed up processing and avoid false positives
+        image = image[self.CROP_V_OFFSET:(self.CROP_V_OFFSET+self.CROP_HEIGHT), self.CROP_H_OFFSET:(self.CROP_H_OFFSET+self.CROP_WIDTH)]
```

Comment:
minor: whitespace around `+`

suggestion: break up this line such that the `self.CROP_V_OFFSET+self.CROP_HEIGHT` and `self.CROP_H_OFFSET+self.CROP_WIDTH` are defined on seperate lines as e.g. `crop_v_limit` & `crop_h_limit` to improve readability